frb_codegen/src/library/codegen/ir/mir/ty/delegate.rs

```diff
@@ -169,9 +180,24 @@ impl MirTypeTrait for MirTypeDelegate {
             MirTypeDelegate::Backtrace => "Backtrace".to_owned(),
             MirTypeDelegate::AnyhowException => "AnyhowException".to_owned(),
             MirTypeDelegate::Map(mir) => {
-                format!("Map_{}_{}", mir.key.safe_ident(), mir.value.safe_ident())
+                if let Some(hasher) = &mir.hasher {
+                    format!(
+                        "Map_{}_{}_{}",
+                        mir.key.safe_ident(),
+                        mir.value.safe_ident(),
+                        hasher.safe_ident(),
+                    )
+                } else {
+                    format!("Map_{}_{}", mir.key.safe_ident(), mir.value.safe_ident())
+                }
```

Comment:
nit: we can unify it by doing something like

                    format!(
                        "Map_{}_{}_{}",
                        mir.key.safe_ident(),
                        mir.value.safe_ident(),
                        hasher.map(|h| h.safe_ident()).unwrap_or("None"),
                    )

etc